File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        Map<Integer, Integer> map = new HashMap<>();
        for( int i=0; i<nums.length; i++ ){

            int required = target  - nums[i];
            Integer  sol = map.get(required);
            if(sol != null){
                return new int[]{i, sol};
            }else{
                map.put(nums[i], i);
            }
            
        }
        return new int[]{-1,-1};
    }
}

File: 1-two-sum/two-sum.go
import (
    "fmt"
)

func twoSum(nums []int, target int) []int {

    dict := make(map[int]int)
    for i:=0; i < len(nums); i++ {
        dict[nums[i]] = i
    }

    for i:=0; i<len(nums); i++ {
        need := target - nums[i]
        if value, ok := dict[need]; ok && value != i{
            return []int{i, dict[need]}
        }
    }

    return []int{}
}